src/events/message/commands/moderation/warn.ts

```diff
@@ -0,0 +1,41 @@
+import { readFileSync } from "fs";
+import * as path from "path";
+import { Role } from "discord.js";
+import { Message, CommandArg } from "../types";
+
+const isModerator = (role: Role) => role.name === "admin" || role.name === "mods";
```

Comment:
> Also, this may be a good time to factor out `isModerator` and `textPath` into a separate file so both `?introduce` and `?warn` commands can reference them. Perhaps `src/common.ts`?

Yeah, `src/common.ts` is fine for now. Something taking `Message` like `fromModerator(message)` is probably more useful, though.

Also, I'd make the `textPath` based on the project root, rather than `process.cwd()`, so it doesn't depend on how you start the bot. I'd also add prefixes to Markdown files because we're using in multiple commands. Maybe `intro-` and `warn-`, but I don't care much as long as it's clear.

Reply:
> Yeah, `src/common.ts` is fine for now. Something taking `Message` like `fromModerator(message)` is probably more useful, though.

Noted with thanks.

> Also, I'd make the `textPath` based on the project root, rather than `process.cwd()`, so it doesn't depend on how you start the bot.

Pardon me for the dumb question, but can we get the project root without using a relative path from the source/compiled file? Or should I use a relative path?

> I'd also add prefixes to Markdown files because we're using in multiple commands. Maybe `intro-` and `warn-`, but I don't care much as long as it's clear.

Would sub-directories work better? E.g. `text/introduce/` and `text/warn/`?

Reply:
> Pardon me for the dumb question, but can we get the project root without using a relative path from the source/compiled file? Or should I use a relative path?

`path.join(__dirname, "../")` in `src/common.ts` should be the absolute path to the project's root. Same result even after compiled to JS `lib/common.js`. See [`__dirname`](https://nodejs.org/docs/latest/api/modules.html#modules_dirname).

> Would sub-directories work better?

Yeah, that's cleaner.